datafusion/src/physical_plan/aggregates.rs

```diff
@@ -40,15 +40,6 @@ use expressions::{
 };
 use std::sync::Arc;
 
-/// the implementation of an aggregate function
```

Comment:
I like that you are slowly disentangling the physical from the logical plans

It seems like eventually we will have "physical plan" depending on Exprs and "logical plan" depending on exprs but not depending on each other 👍

Reply:
> I like that you are slowly disentangling the physical from the logical plans
> 
> It seems like eventually we will have "physical plan" depending on Exprs and "logical plan" depending on exprs but not depending on each other 👍

i had a look but i don't think it's happening anytime soon. now logical plan depends on table scan, which async scan to yield execution plan. it's hard to tear logican plan from execution plan given this chain.